File: app/src/main/java/ru/melod1n/vk/api/model/VKModel.java
package ru.melod1n.vk.api.model;

import org.json.JSONObject;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Collections;

public abstract class VKModel implements Serializable {
    private static final long serialVersionUID = 1L;

    private Object tag;

    protected VKModel() {
    }

    protected VKModel(JSONObject o) {
    }

    public void setTag(Object tag) {
        this.tag = tag;
    }

    public Object getTag() {
        return tag;
    }

    public ArrayList<? extends VKModel> asList() {
        return new ArrayList<>(Collections.singletonList(this));
    }
}

File: app/src/main/java/ru/melod1n/vk/api/model/VKModel.kt
package ru.melod1n.vk.api.model

import org.json.JSONObject
import java.io.Serializable
import java.util.*

abstract class VKModel : Serializable {

    protected constructor()
    protected constructor(o: JSONObject?)

    open fun asList(): ArrayList<out VKModel> {
        return ArrayList(listOf(this))
    }

    companion object {
        private const val serialVersionUID = 1L
    }
}